app/knowledge_panels.py

```diff
@@ -39,3 +41,141 @@ def hunger_game_kp(
+                            "poducts": products[1],
+                            "html": f'<a href="{url[1]}"></a>.',
+                        },
+                        "text_element_3": {
+                            "id": id[2],
+                            "title": name[2],
+                            "known": known[2],
+                            "poducts": products[2],
+                            "html": f'<a href="{url[2]}"></a>.',
+                        },
+                    }
+                ],
+                "source_url": f"{source_url}/data-quality",
+                "description": f"This is a {description}",
+            },
+        }
+    elif len(first_three) == 0:
+        return {
+            "data-quality": {
+                "elements": [
+                    {
+                        "element_type": "text",
+                        "total_issues": total_issues,
+                        "text_element_1": {},
+                    }
+                ],
+                "source_url": f"{source_url}/data-quality",
+                "description": f"This is a {description}",
+            },
+        }
+    elif len(first_three) == 1:
+        return {
+            "data-quality": {
+                "elements": [
+                    {
+                        "element_type": "text",
+                        "total_issues": total_issues,
+                        "text_element_1": {
+                            "id": id[0],
+                            "title": name[0],
+                            "known": known[0],
+                            "poducts": products[0],
+                            "html": f'<a href="{url[0]}"></a>.',
+                        },
+                    }
+                ],
+                "source_url": f"{source_url}/data-quality",
+                "description": f"This is a {description}",
+            },
+        }
+    elif len(first_three) == 2:
+        return {
+            "data-quality": {
+                "elements": [
+                    {
+                        "element_type": "text",
+                        "total_issues": total_issues,
+                        "text_element_1": {
+                            "id": id[0],
+                            "title": name[0],
+                            "known": known[0],
+                            "poducts": products[0],
+                            "html": f'<a href="{url[0]}"></a>.',
+                        },
+                        "text_element_2": {
+                            "id": id[1],
+                            "title": name[1],
+                            "known": known[1],
+                            "poducts": products[1],
+                            "html": f'<a href="{url[1]}"></a>.',
+                        },
+                    }
+                ],
+                "source_url": f"{source_url}/data-quality",
+                "description": f"This is a {description}",
+            },
+        }
```

Comment:
@sumit-158, what happens if we choose to now add 10 items ?

You really have to abstract this out, this can be 3, 4, 5… A loop have to save you here, you should not be unrolling this kind of thing "manually".
You've seen you use the same code with 0, 1, 2… so why not abstract this out with i and an `for i, tag in enumerate(tags)` ?

Moreover you don't have to create more than one element, you should not either introduce new fields to knowledge panels, like known, products, etc...

you should instead have a html with data for every tag, which display useful info to the user

I can do this in one line :-)

```python
html = "\n".join(
    f'<li><a href="{tag.url}">{tag.products} products with {tag.name}</a></li>'  
    for tag in first_three
)
html = f"<ul>{html}</ul>"
```

I let you build the text_element from that.